### DISCUSSION TITLE
using navigator in getInitialProps gives error on build

### DISCUSSION THREAD

**Discussion by OgLuka** [vercel/next.js#39191] [Help]
Hi, I am trying to find the best way to detect the mobile/desktop in Next. 
I read many solutions but nothing seems easy and modern. So the reason I need this is because I need to change my pages layout depending on platform (not possible with CSS only). 

         `  Theplatform.getInitialProps = ({ req }) => {
               let userAgent;
               if (req) { // if you are on the server and you get a 'req' property from your context
               userAgent = req.headers['user-agent'] // get the user-agent from the headers
               } else {
               userAgent = navigator.userAgent // if you are on the client you can access the navigator from the window object
               }
               let isMobileView = Boolean(userAgent.match(
                    /Android|BlackBerry|iPhone|iPad|iPod|Opera Mini|IEMobile|WPDesktop/i
                  ))
                  
                  return { isMobileView }
            } 
        `

this is what I have now and I think It works in development mode but when I build my app it gives me an error that navigator is not defined. of course, I understand the reason because there is no navigator on server-side but shouldn't conditional req fix that?

**Comment by nrooban** [comment]
instead `navigator.userAgent` use `window.navigator.userAgent`
